src/backend/db/database_manager.py

```diff
@@ -850,5 +866,365 @@ def delete_ticket(self, ticket_id, customer_guid):
+                else:
+                    raise Exception("Failed to retrieve the newly created comment ID.")
+
+            session.commit()
+            logger.debug(f"Transaction committed successfully for comment ID {comment_id}")
+
+            return {"comment_id": comment_id, "status": "created"}
+
+        except ValueError as e:
+            logger.error(f"Validation error: {e}")
+            session.rollback()
+            raise e
+
+        except SQLAlchemyError as e:
+            logger.error(f"Database error creating comment for customer {customer_guid}: {e}")
+            session.rollback()
+            raise e
+
+        except Exception as e:
+            logger.error(f"Unexpected error while creating comment: {e}")
+            session.rollback()
+            raise e
+
+        finally:
+            logger.debug("Closing database session")
+            session.close()
+
+    def get_comment_by_id(self, comment_id, customer_guid):
+        customer_db_name = self.get_customer_db(customer_guid)
+        session = DatabaseManager._session_factory()
+        try:
+            # Check if the database exists
+            logger.debug(f"Checking if database {customer_db_name} exists")
+            result = session.execute(
+                text("SELECT SCHEMA_NAME FROM INFORMATION_SCHEMA.SCHEMATA WHERE SCHEMA_NAME = :db_name"),
+                {"db_name": customer_db_name}
+            ).fetchone()
+
+            if not result:
+                raise ValueError(f"Database {customer_db_name} does not exist")
+
+            logger.debug(f"Switching to customer database: {customer_db_name}")
+            session.execute(text(f"USE `{customer_db_name}`"))
+
+            # Retrieve comment details
+            comment_query = '''
+            SELECT comment_id, ticket_id, posted_by, comment, created_at, is_edited
+                FROM ticket_comments
+                WHERE comment_id = :comment_id
+            '''
+            comment_result = session.execute(
+                text(comment_query),
+                {"comment_id": comment_id},
+            ).fetchone()
+
+            if not comment_result:
+                return None  # If comment is not found, return None
+
+            # Convert comment result into a dictionary
+            logger.info(f"Comment details by comment_id {comment_result}")
+            comment_data = {
+                column: value
+                for column, value in zip(comment_result.keys(), comment_result)
+            }
+
+            return comment_data
+
+        except (OperationalError, DatabaseError) as e:
+            logger.error(f"Database error: {e}")
+            raise Exception("Database connectivity issue")
+        except SQLAlchemyError as e:
+            logger.error(f"SQLAlchemy error: {e}")
+            raise Exception("Database query error")
+        finally:
+            session.close()
+
+    def get_comments_by_ticket_id(self, customer_guid, ticket_id):
```

Comment:
Make it a paginated get API

Check for getallchats how it is done

Reply:
modified sir